nautobot/docs/user-guide/administration/upgrading/from-v1/ipam-upgrade-guide.md

```diff
@@ -0,0 +1,191 @@
+
+The "Container" status will be removed and all prefixes will be migrated to the "Active" status if it exists. If the"Active" status was deleted, prefixes will be migrated to the first available prefix status in the database that is not"Container".
+
+### Prefix utilization calculations were revamped
+
+The `get_utilization` method on the `ipam.Prefix` model has been updated in 2.0 to account for the `Prefix.type` field as described above under **Prefix Parenting Guidance**. The behavior is now as follows:
+
+- If the `Prefix.type` is `Container`, the utilization is calculated as the sum of the total address space of all child prefixes.
+- If the `Prefix.type` is `Pool`, the utilization is calculated as the sum of the total number of IP addresses within the pool's range.
+- If the `Prefix.type` is `Network`:
+    - The utilization is calculated as the sum of the total address space of all child `Pool` prefixes plus the total number of child IP addresses.
+    - For IPv4 networks with a `prefix_length` larger (lower) than `/31`, if neither the first or last address is occupied by either a pool or an IP address, they are subtracted from the total size of the prefix.
+
+#### Example
+
+- 192.168.0.0/16          `Container - 1024/65536 utilization`
+    - 192.168.1.0/24      `Network - 1/254 utilization`
+        - 192.168.1.1     `IP Address`
+    - 192.168.2.0/24      `Network - 4/256 utilization`
+        - 192.168.2.0/30  `Pool - 1/4 utilization`
+            - 192.168.2.1 `IP Address`
+    - 192.168.3.0/24      `Network - 5/254 utilization`
+        - 192.168.3.1     `IP Address`
+        - 192.168.3.64/30 `Pool - 0/4 utilization`
+    - 192.168.4.0/24      `Network - 1/256 utilization`
+        - 192.168.4.255   `IP Address`
+
+### IPAddress Parenting Concrete Relationship was added
+
+The `ipam.IPAddress` model has been modified to have a mandatory foreign key to `ipam.Prefix` as the `parent` field. Parenting of IP addresses is now automatically managed at the database level to greatly improve performance especially when calculating tree hierarchy and utilization.
+
+| Removed                | Replaced With   |
+| ---------------------- | --------------- |
+| `get_child_prefixes()` | `descendants()` |
+
+#### IPAddress Parenting Guidance
+
+The following guidance has been added to the `IPAddress` model:
+
+- An `IPAddress` should have a parent `Prefix` of type `Network`
+- An `IPAddress` should not be created if a suitable parent `Prefix` of type `Network` does not exist
+- An `IPAddress` can be a member of a `Pool` but only if the `Pool` is a child of a `Network`
+- If an eligible parent `Prefix` cannot be found for an `IPAddress` in a given `Namespace`, creation or update of that `IPAddress` will fail.
+- If deleting a parent `Prefix` would result in any child `IPAddress` objects to become orphaned, the delete opration will fail.
```

Comment:
These last two items are enforced even in 2.0, right?

Reply:
Yes.